poc/flp_generic.py

```diff
@@ -640,6 +641,137 @@ def test_vec_set_type_param(self, test_vec):
         return ['length', 'chunk_length']
 
 
+class MultiHotHistogram(Valid):
+    """
+    A validity circuit that checks each Client's measurement is a bit vector
+    with at most `max_count` number of 1s.
+
+    In order to check the Client measurement `meas` has at most `max_count` 1s,
+    we ask the Client to send an additional `bits_for_count` bits that
+    encode the number of 1s in the measurement, with an offset. Specifically:
+    - Let `bits_for_count = int(log2(max_count)) + 1`.
+    - Let `offset = 2**bits_for_count - 1 - max_count`.
+    - Client will encode `count = offset + \sum_i meas_i` in
+      `bits_for_count` bits.
+    - We can naturally bound `count` as the following:
+      `0 <= count <= 2**bits_for_count - 1`, and therefore:
+      `-offset <= \sum_i meas_i <= max_count`.
+    - Since we also verify each `meas_i` is a bit, we can lower bound the
+      summation by 0. Therefore, we will be able to verify
+      `0 <= \sum_i meas_i <= max_count`.
+    """
+    # Operational parameters
+    length = None  # Set by the constructor
+    max_count = None  # Set by constructor
+    chunk_length = None  # Set by constructor
+
+    # Associated types
+    Measurement = Vec[Unsigned]
+    AggResult = Vec[Unsigned]
+    Field = field.Field128
+
+    # Associated parameters
+    GADGETS = None  # Set by constructor
+    GADGET_CALLS = None  # Set by constructor
+    MEAS_LEN = None  # Set by constructor
+    JOINT_RAND_LEN = 2
+    OUTPUT_LEN = None  # Set by constructor
+
+    def __init__(self, length, max_count, chunk_length):
+        """
+        Instantiate an instance of the `MultiHotHistogram` circuit with the
+        given length, max_count, and chunk_length.
+        """
+        if length <= 0:
+            raise ValueError('invalid length')
+        if max_count <= 0 or max_count > length:
+            raise ValueError('invalid max_count')
+        if chunk_length <= 0:
+            raise ValueError('invalid chunk_length')
+
+        # Compute the number of bits to represent `max_count`.
+        self.bits_for_count = int(math.log2(max_count)) + 1
```

Comment:
We may as well avoid floating point math when computing this, so we don't have to worry about precision and rounding.
```suggestion
        self.bits_for_count = max_count.bit_length()
```